api/proto/metrics/metrics.proto

```diff
@@ -468,6 +468,43 @@ message ZMetricVolume {
   uint64 freeBytes = 9;
 }
 
+// Metrics for a snapshot
+// When a snapshot is created, its disk space is initially shared between
+// the snapshot and the file system, and possibly with previous snapshots.
+// As the file system changes, disk space that was previously shared becomes
+// unique to the snapshot, and thus is counted in the snapshot's used property.
+// Additionally, deleting snapshots can increase the amount of disk space
+// unique to (and thus used by) other snapshots, however, this does not increase
+// the amount of disk space specified for the volume itself.
+message ZMetricSnapshot {
+  // Snapshot UUID
+  string snapshot_uuid = 1;
+  // User-friendly name on controller
+  string display_name = 2;
+  // Identifies the amount of space (in bytes) consumed by the dataset and all its
+  // descendants. This is a unique value to this snapshot i.e., this is space will
+  // be freed up if the snapshot is deleted.
+  uint64 used_space = 3;
+  // Identifies the amount of data accessible by this snapshot, which might
+  // or might not be shared with other datasets in the pool. When a snapshot or
+  // clone is created, It initially references the same amount of space as the
+  // file system or snapshot it was created from because
+  // its contents are identical. (in bytes)
+  uint64 referenced = 4;
+  // Identifies the compression ratio achieved for this snapshot,
+  // expressed as a multiplier.
+  double compressratio = 5;
+  // Specifies the logical size of the volume this snapshot belongs to (in bytes)
+  uint64 vol_size = 6;
+  // The amount of space that is "logically" accessible by this dataset.
+  // See the referenced property. The logical space ignores the effect of
+  // the compression and copies properties, giving a quantity closer to
+  // the amount of data that applications see.
+  // However, it does include space consumed by metadata.
+  // (in bytes)
+  uint64 logicalreferenced = 7;
```

Comment:
I cannot understand from the comment for `logicalreferenced` what is `the amount of data that applications see` in case of snapshot.
More general question: is it possible to calculate overhead of the particular snapshot at a time? I mean, we have something like current usage and maximum space limit for zvols (volumes). Is there some field that will indicate that snapshot X use N bytes, snapshot Y use M bytes from the persist storage pool at some moment of time?